File: lemongrass-client/src/types/group/GroupResponse.ts
import type { Visibility } from "../enums/visibility.enum";

export interface GroupResponse {
  groupId: string;
  description: string;
  ownerId: string;
  ownerName: string;
  ownerFirstname: string;
  ownerLastname: string;
  category: string;
  name: string;
  memberCount: number;
  coverImageUrl: string;
  rules: string;
  requirePostApproval: boolean;
  visibility: Visibility;
}

File: lemongrass/src/main/java/com/ttoannguyen/lemongrass/dto/response/group/GroupResponse.java
package com.ttoannguyen.lemongrass.dto.Response.group;

import com.ttoannguyen.lemongrass.entity.enums.Visibility;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class GroupResponse {
  String groupId;
  String name;
  String description;
  String ownerId;
  String ownerName;
  String ownerFirstname;
  String ownerLastname;
  String category;
  Visibility visibility;
  String coverImageUrl;
  Integer memberCount;
  String rules;
  boolean requirePostApproval;
}
